www/nginx/src/opnsense/scripts/nginx/setup.php

```diff
@@ -239,11 +239,48 @@ function find_ca($refid)
         unset($file);
     }
 }
+
 // create directories for cache
 foreach ($nginx->cache_path->iterateItems() as $cache_path) {
     @mkdir((string)$cache_path->path, 0755, true);
 }
 
+// create custom error pages
+const ERRORPAGE_DIR = '/usr/local/etc/nginx/views';
+@mkdir(ERRORPAGE_DIR, 0750, true);
+$used_errorpages = array();
+// search used error pages
+foreach ($nginx->http_server->iterateItems() as $http_server) {
+    $pages = explode(',', $http_server->errorpages);
+    foreach ($pages as $page) {
+        $page = str_replace('-', '', $page);
+        if (!in_array($page, $used_errorpages)) {
+            $used_errorpages[] = $page;
+        }
+    }
+}
+// create/update error pages
+foreach ($nginx->errorpage->iterateItems() as $errorpage) {
+    $uuid = str_replace('-', '', $errorpage->getAttributes()['uuid']);
+    if (in_array($uuid, $used_errorpages)) {
+        $filename = "error_$uuid.html";
+        $current_content = @file_get_contents(ERRORPAGE_DIR . "/$filename");
+        $new_content = base64_decode((string)$errorpage->pagecontent);
```

Comment:
I guess it would be better to compare file hashes to you do not have to keep the old file content in memory (just avoid crashes):

use a hash like SHA-256 for the file on disk
* https://www.php.net/manual/de/function.hash-file.php
* https://www.php.net/manual/de/function.hash.php

If you prefer an english page, you can replace the "de" in the URL

Reply:
Good point!